sdk/python/feast/expediagroup/vectordb/elasticsearch_online_store.py

```diff
@@ -71,29 +71,41 @@ class ElasticsearchOnlineStoreConfig(FeastConfigBaseModel):
     password: str
     """ password to connect to Elasticsearch """
 
+    write_batch_size: Optional[int] = 40
+    """ The number of rows to write in a single batch """
 
-class ElasticsearchConnectionManager:
-    def __init__(self, online_config: RepoConfig):
-        self.online_config = online_config
 
-    def __enter__(self):
-        # Connecting to Elasticsearch
-        logger.info(
-            f"Connecting to Elasticsearch with endpoint {self.online_config.endpoint}"
-        )
-        self.client = Elasticsearch(
-            self.online_config.endpoint,
-            basic_auth=(self.online_config.username, self.online_config.password),
+class ElasticsearchOnlineStore(OnlineStore):
+    _client: Optional[Elasticsearch] = None
+
+    def _get_client(self, config: RepoConfig) -> Elasticsearch:
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, ElasticsearchOnlineStoreConfig)
+
+        user = online_store_config.username if online_store_config.username is not None else ""
```

Comment:
can the default values be set in `ElasticsearchOnlineStoreConfig` instead?

Reply:
This is for null safety checks. The config validation is supposed to enforce those values to not be None.